lib/pbench/server/auth/auth.py

```diff
@@ -140,11 +144,10 @@ def verify_third_party_token(auth_token: str, oidc_client: OpenIDClient) -> bool
         Returns:
             True if the verification succeeds else False
         """
-        identity_provider_pubkey = oidc_client.get_oidc_public_key(auth_token)
         try:
             oidc_client.token_introspect_offline(
                 token=auth_token,
-                key=identity_provider_pubkey,
+                key=Auth().get_secret_key(),
```

Comment:
I'm not sure of the intended sequencing here, vis a vis #3094, but it seems disappointing at best to be instantiating and `Auth` object here just so that we can call `get_secret_key()`.